cabal-install/Distribution/Client/Dependency.hs

```diff
@@ -452,35 +444,68 @@ removeLowerBounds (AllowOlder allowNewer)    params =
 
     relaxDeps :: UnresolvedSourcePackage -> UnresolvedSourcePackage
     relaxDeps srcPkg = srcPkg {
-      packageDescription = relaxPackageDeps removeLowerBound allowNewer
+      packageDescription = relaxPackageDeps relKind relDeps
                            (packageDescription srcPkg)
       }
 
 -- | Relax the dependencies of this package if needed.
 --
--- Helper function used by 'removeLowerBound' and 'removeUpperBounds'
-relaxPackageDeps :: (VersionRange -> VersionRange)
+-- Helper function used by 'removeBounds'
+relaxPackageDeps :: RelaxKind
                  -> RelaxDeps
                  -> PD.GenericPackageDescription -> PD.GenericPackageDescription
-relaxPackageDeps _ RelaxDepsNone gpd = gpd
-relaxPackageDeps vrtrans RelaxDepsAll  gpd = PD.transformAllBuildDepends relaxAll gpd
+relaxPackageDeps _ RelaxDepsNone gpd = gpd -- subsumed by no-op case in 'removeBounds'
+relaxPackageDeps relKind RelaxDepsAll  gpd = PD.transformAllBuildDepends relaxAll gpd
   where
-    relaxAll = \(Dependency pkgName verRange) ->
-      Dependency pkgName (vrtrans verRange)
-relaxPackageDeps vrtrans (RelaxDepsSome allowNewerDeps') gpd =
+    relaxAll (Dependency pkgName verRange) =
+        Dependency pkgName (removeBound relKind RelaxDepModNone verRange)
```

Comment:
I wonder if we should allow something like `--allow-newer=^all` to relax only caret dependencies everywhere. Probably not.